### ISSUE TITLE
PropsSI error on Python 3

### ISSUE THREAD

**Issue by spinnau** [ibell/coolprop#181]
While changing the plotting functions to use PropsSI internally, I have found a Python 3 error for PropsSI method (current _master_, example below works on python 2):

``` python
>>> from CoolProp.CoolProp import Props, PropsSI
>>> Props('H', 'T', 400, 'P', 100, 'Water')
2730.427156802709
>>> PropsSI('H', 'T', 400, 'P', 100e3, 'Water')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "CoolProp.pyx", line 355, in CoolProp.CoolProp.PropsSI (build/temp.linux-x86_64-3.3/pyrex/CoolProp/CoolProp.cpp:12165)
  File "CoolProp.pyx", line 447, in CoolProp.CoolProp.PropsSI (build/temp.linux-x86_64-3.3/pyrex/CoolProp/CoolProp.cpp:11562)
  File "CoolProp.pyx", line 351, in CoolProp.CoolProp.__PropsSI_err2 (build/temp.linux-x86_64-3.3/pyrex/CoolProp/CoolProp.cpp:11153)
ValueError: CoolProp error: Your fluid name [Water] is not a CoolProp fluid, a REFPROP fluid, a brine or a liquid :: inputs were:"H","T",4.0000000000000000e+02,"P",1.0000000000000000e+05,"Water"
```

**Comment by ibell** [comment]
Ugh I thought I fixed this problem. I'll fix it first thing tomorrow

**Comment by spinnau** [comment]
Ok, thanks...

**Comment by spinnau** [comment]
@ibell some things with iterable inputs for Props and PropsSI methods went wrong with the last fixes. For both Python 2 and 3 I get

``` python
>>> from CoolProp.CoolProp import Props
>>> import numpy as np
>>> T = np.array([300, 310])
>>> Props('S', 'T', T, 'Q', 0, 'Water')
array([ 0.39308903,  0.53014693])
>>> Props('S', 'Q', 0, 'T', T, 'Water')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "CoolProp.pyx", line 229, in CoolProp.CoolProp.Props (build/temp.linux-x86_64-2.7/pyrex/CoolProp/CoolProp.cpp:10900)
  File "CoolProp.pyx", line 344, in CoolProp.CoolProp.Props (build/temp.linux-x86_64-2.7/pyrex/CoolProp/CoolProp.cpp:10691)
  File "CoolProp.pyx", line 225, in CoolProp.CoolProp.__Props_err2 (build/temp.linux-x86_64-2.7/pyrex/CoolProp/CoolProp.cpp:9816)
ValueError: CoolProp error: Your saturation temperature [0.000000 K] is out of range [273.160000 K, 647.096000 K] :: inputs were:"S","T",3.0000000000000000e+02,"Q",0.0000000000000000e+00,"Water"
```

The order of the arguments does matter. If both inputs are iterables then it works fine.

**Comment by ibell** [comment]
Hmm its supposed to swap them if the iterable is second.  Clearly that's
not happening.  Are you at all comfortable with unit testing?  Could you
add unit tests to for the last two problems you noted to
https://github.com/ibell/coolprop/blob/master/wrappers/Python/CoolProp/tests/test_Props.py
I think the structure should be clear enough.  If you add the unit tests
Ill fix the code.

Ian

**Comment by spinnau** [comment]
I have added the tests and also fixed the problem in PR #182
